cmd/ci-operator-prowgen/main.go

```diff
@@ -91,26 +96,104 @@ func generatePodSpec(org, repo, configFile, target string, additionalArgs ...str
+		template = "cluster-launch-src"
+		clusterType = string(conf.ClusterType)
+	} else if conf := test.OpenshiftInstallerClusterTestConfiguration; conf != nil {
+		template = "cluster-launch-installer-e2e"
+		clusterType = string(conf.ClusterType)
+	} else if conf := test.OpenshiftInstallerSmokeClusterTestConfiguration; conf != nil {
+		template = "cluster-launch-installer-e2e-smoke"
+		clusterType = string(conf.ClusterType)
+	}
+	clusterProfile := fmt.Sprintf("%s-profile", target)
+	clusterProfilePath := fmt.Sprintf("/usr/local/%s", clusterProfile)
+	templatePath := fmt.Sprintf("/usr/local/%s", target)
+	return &kubeapi.PodSpec{
+		ServiceAccountName: "ci-operator",
+		Volumes: []kubeapi.Volume{
+			kubeapi.Volume{
+				Name: "job-definition",
+				VolumeSource: kubeapi.VolumeSource{
+					ConfigMap: &kubeapi.ConfigMapVolumeSource{
+						LocalObjectReference: kubeapi.LocalObjectReference{
+							Name: fmt.Sprintf("prow-job-%s", template)}}}},
+			kubeapi.Volume{
+				Name: "cluster-profile",
+				VolumeSource: kubeapi.VolumeSource{
+					Projected: &kubeapi.ProjectedVolumeSource{
+						Sources: []kubeapi.VolumeProjection{
+							kubeapi.VolumeProjection{
+								Secret: &kubeapi.SecretProjection{
+									LocalObjectReference: kubeapi.LocalObjectReference{
+										Name: fmt.Sprintf("cluster-secrets-%s", clusterType)}}},
+							kubeapi.VolumeProjection{
+								ConfigMap: &kubeapi.ConfigMapProjection{
+									LocalObjectReference: kubeapi.LocalObjectReference{
+										Name: fmt.Sprintf("cluster-profile-%s", clusterType)}}}}}}}},
+		Containers: []kubeapi.Container{
+			kubeapi.Container{
+				Name:      "test",
+				Image:     "ci-operator:latest",
+				Resources: containerResourceReqs,
+				VolumeMounts: []kubeapi.VolumeMount{
+					kubeapi.VolumeMount{
+						Name:      "job-definition",
+						MountPath: templatePath,
+						SubPath:   fmt.Sprintf("%s.yaml", template)},
+					kubeapi.VolumeMount{
+						Name:      "cluster-profile",
+						MountPath: clusterProfilePath}},
+				Env: []kubeapi.EnvVar{
+					kubeapi.EnvVar{
+						Name:  "JOB_NAME_SAFE",
+						Value: strings.Replace(target, "_", "-", -1)},
+					kubeapi.EnvVar{
+						Name:  "CLUSTER_TYPE",
+						Value: clusterType},
+					kubeapi.EnvVar{
+						Name: "CONFIG_SPEC",
+						ValueFrom: &kubeapi.EnvVarSource{
+							ConfigMapKeyRef: &kubeapi.ConfigMapKeySelector{
+								Key: fmt.Sprintf("%s.yaml", branch),
+								LocalObjectReference: kubeapi.LocalObjectReference{
+									Name: fmt.Sprintf("ci-operator-%s-%s", org, repo)}}}},
+					kubeapi.EnvVar{
+						Name:  "TEST_COMMAND",
+						Value: test.Commands},
+					kubeapi.EnvVar{
+						Name:  "RPM_REPO",
+						Value: fmt.Sprintf("https://rpms.svc.ci.openshift.org/openshift-%s/", release)}},
+				Command: []string{
+					"ci-operator",
+					"--artifact-dir=$(ARTIFACTS)",
+					fmt.Sprintf("--secret-dir=%s", clusterProfilePath),
+					fmt.Sprintf("--template=%s", templatePath),
```

Comment:
This needs to be enforced as unique for the org+repo+branch

Reply:
Every occurrence of the string `e2e` — except the job name and the `TEST_COMMAND` — are taken from the name of the test in the ci-operator configuration file:

<details>

```yaml
- agent: kubernetes
  always_run: true
  branches:
  - release-3.10
  context: ci/prow/e2e
  decorate: true
  name: pull-ci-kubernetes-descheduler-release-3.10-e2e
  rerun_command: /test e2e
  skip_cloning: true
  spec:
    containers:
    - command:
      - ci-operator
      - --artifact-dir=$(ARTIFACTS)
      - --secret-dir=/usr/local/e2e-profile
      - --template=/usr/local/e2e
      - --target=e2e
      env:
      - name: JOB_NAME_SAFE
        value: e2e
      - name: CLUSTER_TYPE
        value: gcp
      - name: CONFIG_SPEC
        valueFrom:
          configMapKeyRef:
            key: release-3.10.yaml
            name: ci-operator-kubernetes-descheduler
      - name: TEST_COMMAND
        value: cp /tmp/admin.kubeconfig /tmp/admin.conf && make test-e2e
      - name: RPM_REPO
        value: https://rpms.svc.ci.openshift.org/openshift-origin-v3.10/
      image: ci-operator:latest
      name: test
      resources:
        limits:
          cpu: 500m
        requests:
          cpu: 10m
      volumeMounts:
      - mountPath: /usr/local/e2e
        name: job-definition
        subPath: cluster-launch-src.yaml
      - mountPath: /usr/local/e2e-profile
        name: cluster-profile
    serviceAccountName: ci-operator
    volumes:
    - configMap:
        name: prow-job-cluster-launch-src
      name: job-definition
    - name: cluster-profile
      projected:
        sources:
        - secret:
            name: cluster-secrets-gcp
        - configMap:
            name: cluster-profile-gcp
  trigger: ((?m)^/test( all| e2e),?(\s+|$))
```

</details>